blaise82/l.m.s-client
# #172019567 setup Redux for the whole app

#### What does this PR do?
- This PR setups up Redux
#### Description of Task to be completed?
- Implement boilerplate real-life example that doesn't add value to the LMS to setup and demonstrate the redux flow
- add ```redux``` module
- add ```react-redux``` to read updates from the redux store
- add ```redux-devtools-extension``` for debugging application state changes in the browser
- add ```redux-logge``` middleware to log store in the browser console
- add ```redux-thunker``` middleware to enable action creators to return a function instead of action object
#### How should this be manually tested?
- Clone the repo via ```git clone <repo-link>```
- Run ```npm install``` to install all necessary dependencies
- Run ```npm start``` to start the server locally
- Enable Redux devtools extension in your chrome browser, here is the attached [link](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en) to do so. This will help you to track application state store in the browser
- Inspect the browser to see app state logs created by ```redux-logger``` middleware on fly
- Then test the whole functionality like so. Above all take a look at below screenshots to have a clue on how I tested everything.
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#172019567](https://www.pivotaltracker.com/story/show/172019567)
### Screenshots (if appropriate)
<img width="1135" alt="Screen Shot 2020-03-29 at 19 14 40" src="https://user-images.githubusercontent.com/51243858/77855586-950dfb80-71f1-11ea-8d1a-ee835e9bf5c0.png">
<img width="797" alt="Screen Shot 2020-03-29 at 19 14 15" src="https://user-images.githubusercontent.com/51243858/77855593-9f2ffa00-71f1-11ea-9e5e-49fda69d080c.png">
<img width="776" alt="Screen Shot 2020-03-29 at 19 13 08" src="https://user-images.githubusercontent.com/51243858/77855600-af47d980-71f1-11ea-865f-cf87867cc113.png">
<img width="795" alt="Screen Shot 2020-03-29 at 19 12 46" src="https://user-images.githubusercontent.com/51243858/77855609-bcfd5f00-71f1-11ea-9de1-ecfc3ff35b28.png">
#### Questions:
N/A
